docs/File-System-API.md

```diff
@@ -58,14 +58,69 @@ OutStreamOptions options = new OutStreamOptions.Builder(ClientContext.getConf())
 FileOutStream out = tfs.getOutputStream(path, options);
 ```
 
+### IO Options
+
+There are two different types of storage in Tachyon, Tachyon managed storage and under storage.
+Tachyon managed storage is the memory, SSD, and/or HDD which is allocated to Tachyon workers. For
+read and write operations, users can specify the `TachyonStorageType` to fine tune the interaction
+with Tachyon's managed storage. Likewise, users can specify the `UnderStorageType` to fine tune the
+data operations with the under storage, such as S3, Swift, or HDFS. Note that read operations are
+affected by `TachyonStorageType` (the data can be stored in more Tachyon workers) but not
+`UnderStorageType`.
+
+Below is a table of commonly used `TachyonStorageType` and `UnderStorageType` combinations.
+
+<table class="table">
+<tr><th>TachyonStorageType</th><th>UnderStorageType</th><th>Read Behavior</th><th>Write Behavior</th>
+</tr>
+<tr>
+  <td>PROMOTE</td>
+  <td>NO_PERSIST</td>
+  <td>Data is moved to the highest tier in the worker where the data was read. If the data was not
+  on the local worker, a replica will be added to the local Tachyon worker for each completely read
+  data block.</td>
+  <td>Data is written synchronously to a Tachyon worker.</td>
```

Comment:
Since this is an explanation of `NO_PERSIST`, it could be good to clarify that the data is not written synchronously to the under storage system, e.g. `Data is written synchronously to a Tachyon worker, but not the under storage system`